sklearn/metrics/_pairwise_distances_reduction/_argkmin.pxd.tp

```diff
@@ -0,0 +1,50 @@
+{{py:
+
+implementation_specific_values = [
+    # Values are the following ones:
+    #
+    #       name_suffix, INPUT_DTYPE_t, INPUT_DTYPE
+    #
+    # We also use the float64 dtype and C-type names as defined in
+    # `sklearn.utils._typedefs` to maintain consistency.
+    #
+    ('64', 'DTYPE_t', 'DTYPE'),
```

Comment:
I would be in favor of stopping the use of the confusing `DTYPE_t` and `ITYPE_t` names in newly created source files.

```suggestion
    ('64','cnp.float64_t', 'np.float64'),
```

and delay the work to fix the existing Cython file names in other modules for later clean-up PRs (one per module to keep them manageable to review).

Reply:
I am OK only generating `cnp.float64_t` instead of `DTYPE_t` via Tempita for now and to then perform all the other changes in subsequent PRs, like #24153.

Before performing such change, I think maintainers first need to agree on the types to use for Cython's implementations: in https://github.com/scikit-learn/scikit-learn/pull/23865#discussion_r926716129, @thomasjpfan proposed to use concrete types (e.g. `float` and `double`) directly because numpy redefine them. This way we might be able to get ride of the Cython interfaces for numpy (i.e. `cnp`).

What do you think?

Reply:
Personnally I find the `cnp.float32_t` and `cnp.float64_t` names more explicit and their mapping to matching numpy dtype object more obvious for Cython new-comers.

No strong opinion though.

Reply:
I dont' like using "float" as a type name in discussions because one never knows if it refers to the floating-point type family as a whole (`dtype.kind == "f"`) or as to the specific 32-bit precision floating point representation.

Reply:
So do I, as it can also refer to python `float` which are `PyObject` wrapping C `double`.

I would use numpy types and dtypes for clarity, but I think we choose made the decision after discussions among maintainers working on Cython code. WDYT?